exporter/elasticsearchexporter/integrationtests/exporter_test.go

```diff
@@ -0,0 +1,164 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package integrationtests
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/tidwall/gjson"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/plog/plogotlp"
+	"golang.org/x/sync/errgroup"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+func TestExporter(t *testing.T) {
+	for _, tc := range []struct {
+		name             string
+		restartCollector bool
+		mockESFailure    bool
+	}{
+		{name: "basic"},
+		{name: "es_intermittent_failure", mockESFailure: true},
+		/* Below tests should be enabled after #30792 is fixed
+		{name: "collector_restarts", restartCollector: true},
+		{name: "collector_restart_with_es_intermittent_failure", mockESFailure: true, restartCollector: true},
+		*/
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			runner(t, tc.restartCollector, tc.mockESFailure)
+		})
+	}
+}
+
+func runner(t *testing.T, restartCollector, mockESFailure bool) {
+	t.Helper()
+
+	cfg, err := loadConfig()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	mockES, err := newMockESClient(t, cfg.Debug)
+	require.NoError(t, err)
+
+	collector, err := newTestCollector(t, cfg, mockES.ServerURL)
+	require.NoError(t, err)
+	t.Cleanup(collector.Shutdown)
```

Comment:
I wonder if we could get away with encapsulating this within the `newTestCollector` constructor itself, since that takes `t` as an argument anyway?

Reply:
++, and since we never expect an error, we can have that encapsulated in `newTestCollector` too. So just return a collector that will be shutdown on test cleanup.

Reply:
Thanks for the tips, fixed now: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/31848/commits/f5cf2aeabd77da250c89d2c5a2c802754491a1e0